aiida_testing/mock_code/_fixtures.py

```diff
@@ -175,7 +175,7 @@ def _get_mock_code(
             mock_code_config[label] = code_executable_path
         if code_executable_path not in ('TO_SPECIFY', 'NOT_FOUND') and \
             not pathlib.Path(code_executable_path).is_absolute():
-            code_executable_path = os.fspath(_config.file_path / code_executable_path)
+            code_executable_path = os.fspath(_config.file_path.parent / code_executable_path)
```

Comment:
Just wondering what should be the logical order of these different approaches.

Here is what would make sense to me:

 1. check if it is an absolute path
 2. check if it is a relative path with respect to the config file
 3. check if the executable is in the PATH

Would you agree?

Reply:
Yes I think that is a sensible order 👍 
I have one question about this code part
It starts with 
```python
code_executable_path = mock_code_config.get(label, 'TO_SPECIFY')
if (not code_executable_path) and executable_name:
    code_executable_path = shutil.which(executable_name) or 'NOT_FOUND'
```
If the code is not specified at all it will never take the `executable_name` into account because of the default
So should the executable_name only be relevant if you set the code to an empty string or is this a mistake?

Reply:
Hey @janssenhenning , sorry for the delay.

Indeed, that is a bug and it turns out it was introduced by myself - shame on myself, also for not adding any test for the `executable_name` feature ;-)

I believe my idea would have been that you could forego the need for the config file simply by passing the name of the executable to the pytest fixture and making sure it was in the PATH when the tests are run.

As you point out, at least as the code stands now that would actually only work when using a config file that explicitly sets the code path to an empty value.

I can't remember ever purposefully doing that - given this, I would also be ok with removing the `executable_name` parameter altogether.
I leave it up to you whether you want to make it work or remove it.

Reply:
Just waiting on reply from @janssenhenning before moving forward

Reply:
ping

Reply:
Sorry for being inactive on this. I now adjusted the path determination to this order and added a test for the executable name given case.

I also added two more error messages for the cases, where
- A relative path is given but no executable is found at that path
- no path but the executable name is given and it is not found on the PATH